src/FixtureBuilder/ExpressionLanguage/Lexer/ReferenceEscaperLexer.php

```diff
@@ -44,7 +44,7 @@ public function __construct(LexerInterface $decoratedLexer)
      */
     public function lex(string $value): array
     {
-        $escapedValue = preg_replace('/((\s|^)[a-zA-Z0-9_.+-]+?[A-Za-z0-9])@/', '$1\\@', $value);
+        $escapedValue = preg_replace('/((\s|^)[^@]?(\p{L}|\p{N}|[+-_.])+?(\p{L}|\p{N}))@/', '$1\\@', $value);
```

Comment:
@theofidry Agreed as far as writing more tests goes. I would prefer no auto-escaping than "sometimes works" auto-escaping. You have the final say, this looks like it could create problems with time, so maybe remove auto-escaping with the next major release ?